flow/BlobCipher.h

```diff
@@ -89,51 +81,94 @@ class EncryptBuf : public ReferenceCounted<EncryptBuf>, NonCopyable {
 // This header is persisted along with encrypted buffer, it contains information necessary
 // to assist decrypting the buffers to serve read requests.
 //
-// The total space overhead is 56 bytes.
+// The total space overhead is 96 bytes.
 
 #pragma pack(push, 1) // exact fit - no padding
 typedef struct BlobCipherEncryptHeader {
+	static constexpr int headerSize = 96;
 	union {
 		struct {
 			uint8_t size; // reading first byte is sufficient to determine header
 			              // length. ALWAYS THE FIRST HEADER ELEMENT.
 			uint8_t headerVersion{};
 			uint8_t encryptMode{};
-			uint8_t _reserved[5]{};
+			uint8_t authTokenMode{};
+			uint8_t _reserved[4]{};
 		} flags;
 		uint64_t _padding{};
 	};
-	// Encyrption domain boundary identifier.
-	BlobCipherDomainId encryptDomainId{};
-	// BaseCipher encryption key identifier
-	BlobCipherBaseKeyId baseCipherId{};
-	// Random salt
-	BlobCipherRandomSalt salt{};
-	// Checksum of the encrypted buffer. It protects against 'tampering' of ciphertext as well 'bit rots/flips'.
-	BlobCipherChecksum ciphertextChecksum{};
-	// Initialization vector used to encrypt the payload.
-	uint8_t iv[AES_256_IV_LENGTH];
 
-	BlobCipherEncryptHeader();
+	// Cipher text encryption information
+	struct {
+		// Encyrption domain boundary identifier.
+		EncryptCipherDomainId encryptDomainId{};
```

Comment:
Intention is to make basic data-type definition common which are shared across module; for instance: BaseCipherId, EncryptDomainId etc, modules like BlobCipher, EncryptKeyServer, SimEncryptKeyServer shares them. 

BlobEncyrptHeader is header format specific to BlobCipher encyrption scheme in the proposal, for instance we have : StremCipher and BlobCipher, it is possible in future we can add newer schemes. I thought about it and decided to keep operation specific typedef to retain specific naming properties, whereas, compile defines common across modules into a utility header to avoid any logic bugs or confusion.